### PR TITLE
Add project and test example

### PR DESCRIPTION
**Pull request by bielski**
Example project view is tested by AGSnapshotHelper category for FBSnapshotTestCase.

### CODE REVIEW

File: AGSnapshotHelperExample/AGSnapshotHelperExample/AppDelegate.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+//  Created by Ewa Bielska on 16/01/16.
+//  Copyright © 2016 Allegro Group. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+
+@interface AppDelegate : UIResponder <UIApplicationDelegate>
+
+@property (strong, nonatomic) UIWindow *window;
+
```
Comment (Grubas7) [inline AGSnapshotHelperExample/AGSnapshotHelperExample/AppDelegate.h]: New line not needed.

File: AGSnapshotHelperExample/AGSnapshotHelperExample/AppDelegate.m [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+//
+//  Created by Ewa Bielska on 16/01/16.
+//  Copyright © 2016 Allegro Group. All rights reserved.
+//
+
+#import "AppDelegate.h"
+
+@interface AppDelegate ()
+
+@end
```
Comment (Grubas7) [inline AGSnapshotHelperExample/AGSnapshotHelperExample/AppDelegate.m]: This extension is not needed.

File: AGSnapshotHelperExample/AGSnapshotHelperExample/AppDelegate.m [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    // Called as part of the transition from the background to the inactive state; here you can undo many of the changes made on entering the background.
+}
+
+- (void)applicationDidBecomeActive:(UIApplication *)application {
+    // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
+}
+
+- (void)applicationWillTerminate:(UIApplication *)application {
+    // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
+}
```
Comment (Grubas7) [inline AGSnapshotHelperExample/AGSnapshotHelperExample/AppDelegate.m]: Remove not needed methods.

File: AGSnapshotHelperExample/AGSnapshotHelperExample/ViewController.h [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+//
+//  Created by Ewa Bielska on 16/01/16.
+//  Copyright © 2016 Allegro Group. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+
+@interface ViewController : UIViewController
+
+
```
Comment (Grubas7) [inline AGSnapshotHelperExample/AGSnapshotHelperExample/ViewController.h]: \n

File: AGSnapshotHelperExample/AGSnapshotHelperExample/ViewController.m [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    [super viewDidLoad];
+    // Do any additional setup after loading the view, typically from a nib.
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+@end
```
Comment (Grubas7) [inline AGSnapshotHelperExample/AGSnapshotHelperExample/ViewController.m]: Remove useless code.

File: AGSnapshotHelperExample/AGSnapshotHelperExampleTests/AGSnapshotHelperExampleTests.m [hunk 6]
```diff
@@ -0,0 +1,39 @@
+//
```
Comment (Grubas7) [inline AGSnapshotHelperExample/AGSnapshotHelperExampleTests/AGSnapshotHelperExampleTests.m]: Class needed?

File: Example/AGSnapshotHelperExample/AppDelegate.h [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+//  Copyright © 2016 Allegro Group. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+
+@interface AppDelegate : UIResponder <UIApplicationDelegate>
+
+@property (strong, nonatomic) UIWindow *window;
+
+
```
Comment (Grubas7) [inline Example/AGSnapshotHelperExample/AppDelegate.h]: \n

File: Example/AGSnapshotHelperExample/AppDelegate.m [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+- (void)applicationWillEnterForeground:(UIApplication *)application {
+}
+
+- (void)applicationDidBecomeActive:(UIApplication *)application {
+}
+
+- (void)applicationWillTerminate:(UIApplication *)application {
+}
+
+@end
```
Comment (Grubas7) [inline Example/AGSnapshotHelperExample/AppDelegate.m:13]: Remove not needed methods.

File: Example/AGSnapshotHelperExample/ViewController.h [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+//
+//  Created by Ewa Bielska on 16/01/16.
+//  Copyright © 2016 Allegro Group. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+
+@interface ViewController : UIViewController
+
+
```
Comment (Grubas7) [inline Example/AGSnapshotHelperExample/ViewController.h]: \n

File: Example/AGSnapshotHelperExample/ViewController.m [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+//
+//  Created by Ewa Bielska on 16/01/16.
+//  Copyright © 2016 Allegro Group. All rights reserved.
+//
+
+#import "ViewController.h"
+
+@interface ViewController ()
+
+@end
```
Comment (Grubas7) [inline Example/AGSnapshotHelperExample/ViewController.m]: Extension not needed.

File: Example/AGSnapshotHelperExampleTests/AGSnapshotHelperExampleTests.m [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+- (void)setUp {
+    [super setUp];
+    
+    self.recordMode = NO;
+}
+
+- (void)tearDown {
+    [super tearDown];
+}
```
Comment (Grubas7) [inline Example/AGSnapshotHelperExampleTests/AGSnapshotHelperExampleTests.m]: Method not needed.

File: Example/AGSnapshotHelperExampleTests/AGSnapshotHelperExampleTests.m [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+- (UIView *)initialView {
+    UIStoryboard *storyBoard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
+    ViewController *initialViewController = (ViewController *)[storyBoard instantiateInitialViewController];
```
Comment (Grubas7) [inline Example/AGSnapshotHelperExampleTests/AGSnapshotHelperExampleTests.m]: Casting probably not needed.

File: Example/AGSnapshotHelperExampleTests/AGSnapshotHelperExampleTests.m [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+    
+    self.recordMode = NO;
+}
+
+- (void)tearDown {
+    [super tearDown];
+}
+
+- (UIView *)initialView {
+    UIStoryboard *storyBoard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
```
Comment (Grubas7) [inline Example/AGSnapshotHelperExampleTests/AGSnapshotHelperExampleTests.m]: storyBoard -> storyboard

File: Example/Podfile [hunk 14]
```diff
@@ -0,0 +1,9 @@
+source 'https://github.com/CocoaPods/Specs.git'
+platform :ios, '8.0'
+
+use_frameworks!
```
Comment (Grubas7) [inline Example/Podfile]: Is this needed? AFAIR this is needed only for Swift.

File: Example/AGSnapshotHelperExampleTests/AGSnapshotHelperExampleTests.m (hunk 13 above)
Comment (Grubas7) [inline Example/AGSnapshotHelperExampleTests/AGSnapshotHelperExampleTests.m]: We could name storyboard `Main` more descriptive.

File: Example/Podfile (hunk 14 above)
Comment (Grubas7) [inline Example/Podfile]: Please squash your commits after all:
https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

File: AGSnapshotHelperExample/AGSnapshotHelperExample/AppDelegate.h (hunk 1 above)
Comment (bielski) [inline AGSnapshotHelperExample/AGSnapshotHelperExample/AppDelegate.h]: Removed.

File: AGSnapshotHelperExample/AGSnapshotHelperExample/AppDelegate.m (hunk 3 above)
Comment (bielski) [inline AGSnapshotHelperExample/AGSnapshotHelperExample/AppDelegate.m]: `+1`

File: AGSnapshotHelperExample/AGSnapshotHelperExample/AppDelegate.m (hunk 2 above)
Comment (bielski) [inline AGSnapshotHelperExample/AGSnapshotHelperExample/AppDelegate.m]: Removed.

File: Example/AGSnapshotHelperExampleTests/AGSnapshotHelperExampleTests.m (hunk 12 above)
Comment (bielski) [inline Example/AGSnapshotHelperExampleTests/AGSnapshotHelperExampleTests.m]: `+1`

File: Example/AGSnapshotHelperExampleTests/AGSnapshotHelperExampleTests.m (hunk 13 above)
Comment (bielski) [inline Example/AGSnapshotHelperExampleTests/AGSnapshotHelperExampleTests.m]: I do not find it necessary, but if you insist I can change it in next round.

**Comment by Grubas7** [comment]
Looks OK.

**Pull request merged**